### DISCUSSION TITLE
DevTools: Offline Use Needs a Way to Disable Google Font Link and/or use locally installed fonts

### DISCUSSION THREAD

**Discussion by rdylanwalker** [TanStack/query#6535] [Ideas] [answered]
Hi, I work on an air-gapped network.  The most recent versions of the React Query DevTools auto inserts a link to download Google fonts from the internet, which of course always results in an error.  There is, as far as I can tell, no way to disable this behavior.

I would like the option to turn off this link addition and to either go without the fonts or to use locally installed fonts. I would prefer this be a first-party (vs a fork) option as others may have the same development environment.

**Comment by TkDodo** [comment]
I didn't know that we're using google fonts. We probably shouldn't. @ardeora FYI

**Reply by TkDodo** [reply]
using google fonts violates GDPR:

https://www.patrickriedl.at/wave-of-lawsuits-in-austria-due-to-use-of-google-fonts/

The lawsuits have been dismissed because they sent those letters to hundreds of companies, and it was claimed that they couldn't have visited all those sites by themselves. So at the end, they only tried to make a buck by scaring people into paying.

Still, using google fonts is a big no-no in Europe 😅, and the Munich ruling shows that.

**Reply by rdylanwalker** [reply]
Used here:
packages/query-devtools/src/fonts.ts

**Reply by DamianOsipiuk** [reply]
It's not that we cannot use google fonts, but we should actually bundle them with devtools instead of using cdn.
But ideally we would use some well-known built-in fonts.

**Reply by TkDodo** [reply]
> But ideally we would use some well-known built-in fonts.

agree with that, no need to add our own fonts

**Reply by rdylanwalker** [reply]
Would you like me to open an pull request to remove the function?  

I think the dev tools would fall back what is specified in the CSS of the `body` for the application.

**Reply by TkDodo** [reply]
that would be great 👍

**Reply by rdylanwalker** [reply, marked as answer]
PR: https://github.com/TanStack/query/pull/6573

**Reply by nicholasgriffintn** [reply]
Thanks for this, was just about to raise an issue myself as this has been bugging me (CSP blocking them)
